File: tata-fne-back-end/src/main/java/com/bzdata/TataFneBackend/newCertificationWay/InvoiceSignRequest.java
package com.bzdata.TataFneBackend.newCertificationWay;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;

import java.util.List;

@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class InvoiceSignRequest {

    private String invoiceType;        // "sale"
    private String paymentMethod;      // "transfer"
    private String template;           // "B2B"
    private String numeroFacture;
    private String clientNcc;
    private String clientCompanyName;
    private String clientPhone;
    private String clientEmail;

    private String pointOfSale;
    private String establishment;

    private String commercialMessage;

    private List<InvoiceItem> items;

    private Integer discount;          // 20
}

File: tata-fne-front-end/src/app/core/models/invoice-sign-request.ts
export interface InvoiceItem {
  taxes?: string[];
  description?: string;
  quantity?: number;
  amount?: number;
}

export interface InvoiceSignRequest {
  invoiceType?: string;
  paymentMethod?: string;
  template?: string;
  numeroFacture?: string;
  clientNcc?: string;
  clientCompanyName?: string;
  clientPhone?: string;
  clientEmail?: string;
  pointOfSale?: string;
  establishment?: string;
  commercialMessage?: string;
  items?: InvoiceItem[];
  discount?: number;
}
